ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/generators/clientmodels/BaseClientGenerator.java

```diff
@@ -37,8 +39,41 @@
     public BaseClient generate(final ImmutableList<Ds3Request> ds3Requests) {
         final ImmutableList<PayloadCommand> payloadCommands = toPayloadCommands(ds3Requests);
         final ImmutableList<VoidCommand> voidCommands = toVoidCommands(ds3Requests);
+        final ImmutableList<SpecializedCommand> specializedCommands = toSpecializedCommands(ds3Requests);
 
-        return new BaseClient(payloadCommands, voidCommands);
+        return new BaseClient(payloadCommands, voidCommands, specializedCommands);
+    }
+
+    /**
+     * Creates the list of specialized commands for the client
+     */
+    protected static ImmutableList<SpecializedCommand> toSpecializedCommands(final ImmutableList<Ds3Request> ds3Requests) {
+        if (isEmpty(ds3Requests)) {
+            return ImmutableList.of();
+        }
+        final ImmutableList.Builder<SpecializedCommand> builder = ImmutableList.builder();
```

Comment:
This can be simplified to:
```java
ds3Requests.stream()
    .filter(this::isGetObjectAmazonS3Request)
    .map(this::toGetObjectCommand)
    .collect(GuavaCollectors.immutableList());
```